File: DI/DI-Dagger-Java/src/main/java/DI/cafetera/CafeterasComponent.java
package DI.cafetera;

import dagger.Component;
import model.cafetera.Cafetera;

import javax.inject.Singleton;

@Singleton // Siempre es la misma instancia.
// Los módulos que vamos a usar
@Component(modules = {
        CalentadorModule.class,
        BombaModule.class
})
public interface CafeterasComponent {
    Cafetera build();
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/DI/cafeteras/CafeterasComponent.kt
package DI.cafeteras

import dagger.Component
import models.cafeteras.Cafetera

import javax.inject.Singleton


@Singleton // Siempre es la misma instancia.
// Los módulos que vamos a usar
@Component(modules = [CalentadoresModule::class, BombasModule::class])
interface CafeterasComponent {
    fun build(): Cafetera
}
